### PR TITLE
Preserve column order

### LINKED ISSUE #1594: DefaultMapper should return maps with predictable order
I've came accross this issue while writing unit tests with asserts on SQL query strings. I've noticed that between multiple runs of the same test, the order of fields in generated SQLs varied. This rendered my tests flaky. 
Also, because of this, we could possibly be expecting different SQL queries for the same lists of fields produced on different JVMs in the production environment. This in turn would decrease database's ability to skip parsing similar statements.
The quickest fix would be to replace HashMap with a LinkedHashMap in DefaultMapper.

Thanks in advance

### PR DESCRIPTION
**Pull request by johnktims**
Fixes #1594
Backport of #1597

### CODE REVIEW

File: querydsl-sql/src/test/java/com/mysema/query/sql/dml/SQLInsertClauseTest.java [hunk 1]
```diff
@@ -20,4 +22,17 @@ public void GetSQL() {
         assertEquals(ImmutableList.of(1), sql.getBindings());
     }
 
+    @Test
+        public void GetSQLWithPreservedColumnOrder() {
```
Comment (timowest) [inline querydsl-sql/src/test/java/com/mysema/query/sql/dml/SQLInsertClauseTest.java]: indentation is off here

Comment (Shredder121) [inline querydsl-sql/src/test/java/com/mysema/query/sql/dml/SQLInsertClauseTest.java]: Haha, you think so?

Comment (Shredder121) [inline querydsl-sql/src/test/java/com/mysema/query/sql/dml/SQLInsertClauseTest.java]: @johnktims do you want me to cherry-pick the original commit?

Comment (johnktims) [inline querydsl-sql/src/test/java/com/mysema/query/sql/dml/SQLInsertClauseTest.java]: Yes.

**Comment by Shredder121** [comment]
There you go!

**Comment by johnktims** [comment]
Thanks :)

**Comment by Shredder121** [comment]
One second.

File: querydsl-sql/src/test/java/com/mysema/query/sql/dml/SQLInsertClauseTest.java [hunk 2]
```diff
@@ -20,4 +20,17 @@ public void GetSQL() {
         assertEquals(ImmutableList.of(1), sql.getBindings());
     }
 
+    @Test
+    public void GetSQLWithPreservedColumnOrder() {
+        com.querydsl.sql.domain.QEmployee emp1 = new com.querydsl.sql.domain.QEmployee("emp1");
```
Comment (Shredder121) [inline querydsl-sql/src/test/java/com/mysema/query/sql/dml/SQLInsertClauseTest.java]: Do you want me to update the fully qualified name or update the import of `QEmployee`?

**Comment by timowest** [comment]
This could be maybe also backported in this PR in https://github.com/querydsl/querydsl/pull/1604

Otherwise the order isn't defined, just dependent on less variables than before.

**Comment by Shredder121** [comment]
That's fine, I'll do that.

**Pull request merged**